src/vanilla/internals.ts

```diff
@@ -780,38 +788,26 @@ export const INTERNAL_getStoreStateRev1 = (store: unknown): StoreState =>
   (store as Store)[STORE_STATE]!
 
 export const INTERNAL_buildStore = (...storeArgs: StoreArgs): Store => {
```

Comment:
jotai-scope originally needed unstable_derive to reuse the atomState map from the parent store. AFAIK, that may still be necessary.

How about this?
```suggestion
export const INTERNAL_buildStore = (
  storeArgs: StoreArgs,
  [
    storeHooks = {},
    mountedAtoms = new WeakMap<AnyAtom, Mounted>(),
    invalidatedAtoms = new WeakMap<AnyAtom, EpochNumber>(),
    changedAtoms = new Map<AnyAtom, INTERNAL_AtomState>(),
    mountCallbacks = new Set<() => void>(),
    unmountCallbacks = new Set<() => void>(),
    atomStateMap = new WeakMap<AnyAtom, AtomState>(),
  ]: StoreStates
): Store => {
```

Reply:
> I'm not sure but I think that may still be necessary.

It's still possible as we don't lose any capabilities, but I like your approach. Let me try something.

Reply:
73a394d I intentionally avoided having defaults for now.